### DISCUSSION TITLE
Output when using Prog device

### DISCUSSION THREAD

**Discussion by samrausch** [artisan-roaster-scope/artisan#1033] [Q&A] [answered]
I'm running Artisan on an RPi4 with TC connected directly to the GPIO.  I have my Device set to Prog and running a Python script that returns the values of the TC's, which is working great for the graphing/logging.

I'm now trying to figure out how to have Artisan output data to another Python script for further processing, but there's no documentation I can find about what Artisan does when you check the Output box under Prog.  Before I start writing something in Python to reverse engineer the output, is there any info on how to use this option?

**Comment by MAKOMO** [comment, marked as answer]
The Output script is called (if ticked) even if Prog device is not used. Artisan calls the given script with 4 arguments: ET, BT, ET Background, BT Background.

A simple python script taking the 4 values and appending them to /tmp/out.txt could look like this:

```python
#!/usr/bin/python3

import sys
ET, BT, ETB, BTB = sys.argv[1:]

with open("/tmp/out.txt", "w+") as file:
  file.write(f'ET: {ET}, BT: {ET}, ETB: {ETB}, BTB: {BTB}\n')
```

or similar as a shell script

```bash
#!/bin/bash
echo "ET: $1" >> /tmp/test.out
echo "BT: $2" >> /tmp/test.out
echo "ETB: $3" >> /tmp/test.out
echo "BTB: $4" >> /tmp/test.out
echo "---" >> /tmp/test.out
```

Note that the script is called in an empty environment thus relative paths might not work.

Note that there is also the Call Program event action which can call scripts.
